### PR TITLE
Allow to customize fallback message

### LINKED ISSUE #9: Allow to customize fallback message
Message template applies to text field, but not fallback. Preferrably, allow to customize separately.

From the slack docs:

> Please note that the fallback field is required, and is displayed whenever message attachments cannot be shown (ie. mobile notifications, desktop notifications, IRC).

### PR DESCRIPTION
**Pull request by wesleimp**
Close #9

### CODE REVIEW

**Comment by wesleimp** [comment]
@appleboy can your merge this pr?

File: plugin.go [hunk 1]
```diff
@@ -45,15 +45,16 @@ type (
...
-		ImageURL  string
-		IconURL   string
-		IconEmoji string
-		LinkNames bool
+		Webhook          string
+		Channel          string
+		Recipient        string
+		Username         string
+		Template         string
+		TemplateFallback string
```
Comment (bradrydzewski) [inline plugin.go]: I suggest we just call this `Fallback`

**Comment by bradrydzewski** [comment]
this is missing the `PLUGIN_FALLBACK` environment variable configuration. Without defining this in the flags section, you will not be able to configure the fallback message in your yaml.

**Comment by wesleimp** [comment]
@bradrydzewski sorry about the flags, my bad. 

I think now the changes are ok.

File: main.go [hunk 2]
```diff
@@ -46,6 +46,11 @@ func main() {
 			Usage:  "slack template",
 			EnvVar: "PLUGIN_TEMPLATE",
 		},
+		cli.StringFlag{
+			Name:   "fallback",
+			Usage:  "slack fallback",
+			EnvVar: "PLUGIN_FALLBACK",
+		},
```
Comment (bradrydzewski) [inline main.go]: you need to get the value and pass to the configuration object here: https://github.com/drone-plugins/drone-slack/blob/6bd3a07ba87000bc85932641fb497f9bf377f2a0/main.go#L220:L229

File: plugin_test.go [hunk 3]
```diff
@@ -1,7 +1,10 @@
 package main
 
-import "testing"
-import "gotest.tools/assert"
+import (
+	"testing"
+
+	"gotest.tools/assert"
```
Comment (bradrydzewski) [inline plugin_test.go]: also please remove this dependency.  We use the Go standard library only for testing.

Comment (bradrydzewski) [reply plugin_test.go]: oh nevermind, it looks like someone else added this library to the repository.  This should not have slipped through code review, but we will remove this in a separate PR.

Comment (wesleimp) [reply plugin_test.go]: I think the `go fmt` brings that to the code review

**Comment by wesleimp** [comment]
@bradrydzewski all the requested changes are done

File: plugin.go [hunk 4]
```diff
@@ -90,11 +91,19 @@ func (m Message) String() string {
 
 func (p Plugin) Exec() error {
 	attachment := slack.Attachment{
-		Fallback:   fallback(p.Repo, p.Build),
 		Color:      color(p.Build),
 		MarkdownIn: []string{"text", "fallback"},
 		ImageURL:   p.Config.ImageURL,
 	}
+	if p.Config.Fallback != "" {
+		var err error
```
Comment (tboerger) [inline plugin.go]: I would prefer something like this:

```
a, err := templateMessage(p.Config.Fallback, p)

if err != nil {
    return err
}

attachment.Fallback = a
```

Comment (wesleimp) [reply plugin.go]: Done

**Pull request merged**